frame/nomination-pools/src/lib.rs

```diff
@@ -582,25 +583,237 @@ pub struct PoolRoles<AccountId> {
+	/// is present, record the current block  as the previously updated commission. If the supplied
+	/// commission is zero, `None` will be  inserted and `payee` will be ignored.
+	fn maybe_update_current(
+		&mut self,
+		current: &Option<(Perbill, T::AccountId)>,
+	) -> DispatchResult {
+		if current.is_none() {
+			self.current = None;
+		} else {
+			let (commission, payee) = current.as_ref().ok_or(Error::<T>::NoCommissionSet)?;
+			ensure!(!self.throttling(&commission), Error::<T>::CommissionChangeThrottled);
+			ensure!(
+				self.max.map_or(true, |m| commission <= &m),
+				Error::<T>::CommissionExceedsMaximum
+			);
+			self.current = Some((*commission, payee.clone()));
+		}
+		let _ = self.register_update(<frame_system::Pallet<T>>::block_number());
+		Ok(())
+	}
+
+	/// Set the pool's maximum commission.
+	///
+	/// The pool's maximum commission can initially be set to any value, below global maximum if
+	/// set, and only smaller values thereafter. If larger values are attempted, this function will
+	/// return a dispatch error.
+	///
+	/// If `current.0` is larger than an updated max commission value, then `current.0` will also be
+	/// updated to the new maximum. This will also register a `last_updated` update to the
+	/// commission.
+	fn maybe_update_max(&mut self, new_max: Perbill) -> DispatchResult {
+		if let Some(old) = self.max.as_mut() {
+			if new_max > *old {
+				return Err(Error::<T>::MaxCommissionRestricted.into())
+			}
+			*old = new_max;
+		} else {
+			self.max = Some(new_max)
+		};
+		let updated_current = self
+			.current
+			.as_mut()
+			.map(|(x, _)| {
+				let u = *x > new_max;
+				*x = (*x).min(new_max);
+				u
+			})
+			.unwrap_or(false);
+
+		if updated_current {
+			self.register_update(<frame_system::Pallet<T>>::block_number());
+		}
+		Ok(())
+	}
+
+	/// Set the pool's commission `change_rate` settings.
+	///
+	/// Once a change rate configuration has been set, only more restrictive values can be set
+	/// thereafter. These restrictions translate to increased `min_delay` values and decreased
+	/// `max_increase` values.
+	fn maybe_update_change_rate(
+		&mut self,
+		change_rate: CommissionChangeRate<T::BlockNumber>,
+	) -> DispatchResult {
+		ensure!(&self.less_restrictive(&change_rate), Error::<T>::CommissionChangeRateNotAllowed);
+		self.change_rate = Some(change_rate);
+
+		Ok(())
+	}
+
+	/// Gets the current commission (if any) and payee to be paid.
+	///
+	/// Commission cannot go beyond `GlobalMaxCommission`. A zero commission along with a `None`
+	/// payee is returned in the event a commission has not been configured to the pool. `None` is
+	/// returned in the event a commission has not been configured to the pool.
+	fn get_commission_and_payee(
+		&self,
+		pending_rewards: &BalanceOf<T>,
+	) -> Option<(BalanceOf<T>, T::AccountId)> {
+		self.current.as_ref().map(|(commission, payee)| {
+			(
+				*commission.min(&GlobalMaxCommission::<T>::get().unwrap_or(Bounded::max_value())) *
```

Comment:
@kianenigma the commission is bounded by the global max here, this is where the final commission is determined & applied to a payout. With this `GlobalMaxCommission` does not need to play a part in local pool commission updates.